tests/countries/test_yemen.py

```diff
@@ -0,0 +1,237 @@
+
+    def test_hijri_new_year(self):
+        name = "عيد رأس السنة الهجرية"
+        dts = (
+            "2020-08-20",
+            "2021-08-10",
+            "2022-07-30",
+            "2023-07-19",
+            "2024-07-07",
+        )
+        self.assertHolidayName(name, dts)
+        self.assertHolidayName(name, self.no_estimated_holidays, range(1991, 2050))
+
+    def test_mawlid(self):
+        name = "المولد النبوي"
+        dts = (
+            "2020-10-29",
+            "2021-10-18",
+            "2022-10-08",
+            "2023-09-27",
+            "2024-09-15",
+        )
+        self.assertHolidayName(name, self.workday_no_estimated_holidays, dts)
+        self.assertHolidayName(name, self.workday_no_estimated_holidays, range(2000, 2050))
+
+    def test_isra_and_miraj(self):
+        name = "ذكرى الإسراء والمعراج"
+        dts = (
+            "2023-02-18",
+            "2024-02-08",
+            "2025-01-27",
+        )
+        self.assertHolidayName(name, self.workday_no_estimated_holidays, dts)
+        self.assertHolidayName(name, self.workday_no_estimated_holidays, range(2000, 2050))
+
+    def test_eid_al_fitr(self):
+        name = "عيد الفطر"
+        dts = (
+            "2020-05-23",
+            "2020-05-24",
+            "2020-05-25",
+            "2020-05-26",
+            "2021-05-12",
+            "2021-05-13",
+            "2021-05-14",
+            "2021-05-15",
+            "2022-05-01",
+            "2022-05-02",
+            "2022-05-03",
+            "2022-05-04",
+            "2023-04-20",
+            "2023-04-21",
+            "2023-04-22",
+            "2023-04-23",
+            "2024-04-08",
+            "2024-04-09",
+            "2024-04-10",
+            "2024-04-11",
+            "2024-04-12",
+            "2025-03-29",
+            "2025-03-30",
+            "2025-03-31",
+            "2025-04-01",
+        )
+        self.assertHolidayName(name, dts)
+        self.assertHolidayName(name, self.no_estimated_holidays, range(1991, 2050))
+
+    def test_eid_al_adha(self):
+        name = "عيد الأضحى"
+        dts = (
+            "2020-07-30",
+            "2020-07-31",
+            "2020-08-01",
+            "2020-08-02",
+            "2020-08-03",
+            "2021-07-19",
+            "2021-07-20",
+            "2021-07-21",
+            "2021-07-22",
+            "2021-07-23",
+            "2022-07-08",
+            "2022-07-09",
+            "2022-07-10",
+            "2022-07-11",
+            "2022-07-12",
+            "2023-06-27",
+            "2023-06-28",
+            "2023-06-29",
+            "2023-06-30",
+            "2023-07-01",
+            "2024-06-15",
+            "2024-06-16",
+            "2024-06-17",
+            "2024-06-18",
+            "2024-06-19",
+        )
+        self.assertHolidayName(name, dts)
+        self.assertHolidayName(name, self.no_estimated_holidays, range(1991, 2050))
+
+    def test_observed(self):
+        dts = (
```

Comment:
Observed holidays checks should be placed in holidays test methods, for each holiday separately.